### DISCUSSION TITLE
How to implement hooks in the extension file

### DISCUSSION THREAD

**Discussion by Ivann-2612** [directus/directus#13561] [Archive]
https://docs.directus.io/extensions/hooks/#custom-api-hooks
Hello, 
I have one question, I have made a form that sends data to Directus using the post method. How can I send an email to this address to thank the user from this information, including the email address?
On submitting the form and after receiving data on Directus to send the user an email.
The trigger would be the submit button and the data goes through the smtp server set in the .env Directus file.
each user when filling out the form data is sent to Directus.
I want to catch that user's email and return the email.

**Comment by rijkvanzanten** [comment]
> On submitting the form and after receiving data on Directus to send the user an email.

You would use an `action` hook listening to `items.create` 👍🏻

**Reply by Ivann-2612** [reply]
import axios from 'axios';

export default ({ filter,action }, { services, exceptions }) => {
	const { MailService } = services;
	const { ServiceUnavailableException, ForbiddenException } = exceptions;
    console.log(services, exceptions);

	// Sync with external recipes service, cancel creation on failure
	filter('items.read', async (input, { collection }, { schema }) => {
		if (collection !== 'user') return input;
        console.log(collection);

		const mailService = new MailService({ schema });

		try {
			await axios.post('http://localhost:8055/items/Contact_Form', input);
			await mailService.send({
				to: 'ivan.nikolicamp@gmail.com',
				template: {
					name: 'user-created',
					data: {
						collection: collection,
					},
				},
			});
		} catch (error) {
			throw new ServiceUnavailableException(error);
		}

		input.syncedWithExample = true;

		return input;
	});
    action('items.create', () => {
		console.log('Item created!');
	});
};
Is this ok or something else?

**Reply by Ivann-2612** [reply]
if you have a file or an example where i can see how it works,i am confussed

**Reply by Ivann-2612** [reply]
![Screenshot (289)](https://user-images.githubusercontent.com/60244706/170376240-e52cd3db-008c-46ca-8958-9e23289cd114.png)
This is my Directus folder

**Reply by Ivann-2612** [reply]
And my endpoint is localhost:8055/items/Contact-form where the data from the form from the frontend is stored with POST method
